### PR TITLE
Solved issue 63

### LINKED ISSUE #63: Enhancement - add fee support
Currently, fees are excluded from all calculations. Things that must be done include: 

- Determine whether the fee  in `config.json` must be a percentage or fraction? Or can be both, and percentage is specified by typing for instance `fee: "0.1%"`
- Come up with suitable validations for the provided fee value (e.g., must be positive, less than 1 or 100%)
- Fees must be included in budget changes in `Tradingmodule.open_trade()` and `.close_trade()`.
- Save fee statistics like `total_fee_paid` in `TradingModule`, and print them in the final results (`BackTesting`).

### PR DESCRIPTION
**Pull request by UdristeAndrei**
close #63 Added and calculated the fee_amount and the total_fee_ammount based on the fee imposed by the client. In case the fee is invalid we will use a default fee. Added a fee verification function. Print the total_fee_amount in the BackTesting file.

### CODE REVIEW

File: data/tradingmodule.py [hunk 1]
```diff
@@ -30,13 +33,37 @@ class TradingModule:
...
+    total_fee_amount = 0
+
     def __init__(self, config):
         print("[INFO] Initializing trading-module")
         self.config = config
 
         self.strategy = load_strategy_from_config(config)
         self.budget = float(self.config['starting-capital'])
         self.max_open_trades = int(self.config['max-open-trades'])
+        self.fee = self.verify_fee(self.config['fee'])
```
Comment (noudcorten) [inline data/tradingmodule.py:46]: Add division by 100 here instead of doing it on line 148 & 173

Comment (dema-trading-ai) [reply data/tradingmodule.py:46]: @noudcorten good remark.

File: data/tradingmodule.py [hunk 2]
```diff
@@ -30,13 +33,37 @@ class TradingModule:
...
+            print(
+                "The imputed value is to big, the alforithm will use the default value of {DEFAULT_FEE}%")
+            return DEFAULT_FEE
+        elif input_fee < MIN_FEE:
+            print(
+                "The imputed value is to small, the alforithm will use the default value of {DEFAULT_FEE}%")
+            return DEFAULT_FEE
+
+        print(f"The alforithm will use the imputed value of {input_fee}%")
+        return input_fee
```
Comment (noudcorten) [inline data/tradingmodule.py:66]: Wouldn't it be more logical to round the input_fee to the nearest defined fee? So basically in pseudocode:
* `if fee is defined:`
    * `if defined_fee < min_fee: defined_fee = min_fee`
    * `elif defined_fee > max_fee: definded_fee = max_fee`
* `elif fee is not defined:`
    * `defined_fee = default_fee`

File: data/tradingmodule.py [hunk 3]
```diff
@@ -30,13 +33,37 @@ class TradingModule:
...
         self.budget = float(self.config['starting-capital'])
         self.max_open_trades = int(self.config['max-open-trades'])
+        self.fee = self.verify_fee(self.config['fee']) / 100
+
+    def verify_fee(self, fee):
+        try:
+            input_fee = float(fee)
+        except ValueError:
+            print(
+                f"The imputed value is invalid, the alforithm will use the default value of {DEFAULT_FEE}%")
```
Comment (noudcorten) [inline data/tradingmodule.py:53]: Misspelled two words:
- 'imputed' -> 'inputted' (also on line 58, 62 & 65)
- 'alforithm' -> 'algorithm' (also on line 58, 62 & 65)

**Review by dema-trading-ai — APPROVED** [review]

**Comment by dema-trading-ai** [comment]
@noudcorten what do you think? This is now merge-able right? could you approve if so

File: data/tradingmodule.py [hunk 4]
```diff
@@ -30,13 +33,38 @@ class TradingModule:
...
     def __init__(self, config):
         print("[INFO] Initializing trading-module")
         self.config = config
 
         self.strategy = load_strategy_from_config(config)
         self.budget = float(self.config['starting-capital'])
         self.max_open_trades = int(self.config['max-open-trades'])
+        self.fee = self.verify_fee(self.config['fee']) / 100
+
+    def verify_fee(self, fee):
```
Comment (pvdklei) [inline data/tradingmodule.py:48]: Maybe it's nicer to add this to the config validations, instead of in the TradingModule.

**Pull request merged**